Packages/ClientRuntime/Sources/Waiters/WaiterScheduler.swift

```diff
@@ -0,0 +1,85 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+/// Calculates the time for the next retry, and whether waiting should continue.
+/// Upon creation, the scheduler will allow an immediate request.
+/// After the first request, further requests are scheduled per the retry strategy formula
+/// published in the Smithy specification:
+/// https://awslabs.github.io/smithy/2.0/additional-specs/waiters.html#waiter-retries
+class WaiterScheduler {
```

Comment:
This not how we should schedule. What is happening now

We have a standard retry strategy applied. That is going to try 3 times (I think that's we have configured) withe exponential backoff with jitter. On top that this scheduler works.

Rather we want to use the config provided as part of the waiter and apply them to the original retry strategy.

Reply:
Per our discussion, keeping waiter retry as its own mechanism that "wraps" the client & operation.